crypto/x509/x509_vfy.c

```diff
@@ -3306,14 +3303,15 @@ static int build_chain(X509_STORE_CTX *ctx)
     }
 
  int_err:
-    sk_X509_free(sk_untrusted);
     ERR_raise(ERR_LIB_X509, ERR_R_INTERNAL_ERROR);
     ctx->error = X509_V_ERR_UNSPECIFIED;
-    return -1;
+    goto free;
 
  memerr:
     ERR_raise(ERR_LIB_X509, ERR_R_MALLOC_FAILURE);
     ctx->error = X509_V_ERR_OUT_OF_MEM;
+ free:
+    sk_X509_free(sk_untrusted);
```

Comment:
This is seriously ugly construction. What about just having `sk_X509_free(sk_untrusted);` in both int_err and memerr goto targets?

Reply:
Hmm, having 
```
    sk_X509_free(sk_untrusted);
    return -1;
```
in both cases add redundancy, but just a minor one, so done this way.